jedi/api/__init__.py

```diff
@@ -108,7 +110,11 @@ def __init__(self, source=None, line=None, column=None, path=None,
         self._user_context = UserContext(self.source, self._pos)
         self._parser = UserContextParser(self._grammar, self.source, path,
                                          self._pos, self._user_context)
-        self._evaluator = Evaluator(self._grammar)
+        if sys_path is None:
+            venv = os.getenv('VIRTUAL_ENV')
+            if venv:
+                sys_path = list(get_venv_path(venv)) + sys.path
```

Comment:
Hmm I've read it, very good article! I think I finally understand why `site.py` exists and what it does.

If you start a new process with Jedi in an activated virtualenv, wouldn't we see the correct `sys.path`? Are you talking about a changing just the  `VIRTUAL_ENV` variable without restarting the process?

In general, have you read the idea of #385, to make virtualenv support easier?